pkg/util/globalerror/errors.go

```diff
@@ -47,7 +48,10 @@ const (
 	MetricMetadataHelpTooLong       ID = "help-too-long"
 	MetricMetadataUnitTooLong       ID = "unit-too-long"
 
-	MaxQueryLength ID = "max-query-length"
+	MaxQueryLength       ID = "max-query-length"
+	RequestRateLimited   ID = "request-rate-limited"
+	IngestionRateLimited ID = "ingestion-rate-limited"
```

Comment:
I would change them to match DistributorMaxIngestionRate, IngesterMaxIngestionRate, etc...

```suggestion
	RequestRateLimited   ID = "max-request-rate"
	IngestionRateLimited ID = "max-ingestion-rate"
```

Reply:
Re-thinking about this, we could add the "tenant-" prefix to clarify these are per-tenant limits (similarly to the "distributor-" and "ingester-" prefix we used for the per-instance limits):

```suggestion
	RequestRateLimited   ID = "tenant-max-request-rate"
	IngestionRateLimited ID = "tenant-max-ingestion-rate"
```